### PR TITLE
Update ZIO-Kafka version

### PR DESCRIPTION
**Pull request by yisraelU**
this will upgrade kafka client version and remove [CVE](https://kafka.apache.org/cve-list.html) related to earlier Kafka Clients

### CODE REVIEW

**Review by soujiro32167 — APPROVED** [review]

**Comment by yisraelU** [comment]
Well apparently the interface was changed, so this is no longer a simple single character PR
Will see if I can come up with something

**Comment by soujiro32167** [comment]
Thanks @yisraelU ! I'll take a look

File: zio-kafka/src/main/scala/io/kaizensolutions/trace4cats/zio/extras/ziokafka/KafkaProducerTracer.scala [hunk 1]
```diff
@@ -100,8 +132,11 @@ object KafkaProducerTracer {
                    .ignore
             kafkaTraceHeaders =
               traceHeaders.values.map { case (k, v) => new RecordHeader(k.toString, v.getBytes) }.toList
-            recordsWithHeaders <- ZIO.foreach(records)(addHeaders(kafkaTraceHeaders, _))
-            waitForAck         <- underlying.produceChunkAsync(recordsWithHeaders, keySerializer, valueSerializer)
+            recordsWithHeaders <-
+              ZIO.foreach(records)(
+                addHeaders(kafkaTraceHeaders, _).flatMap(serialize(_, keySerializer, valueSerializer))
```
[the remark this answers is not in the document]
Comment (yisraelU) [reply zio-kafka/src/main/scala/io/kaizensolutions/trace4cats/zio/extras/ziokafka/KafkaProducerTracer.scala:137]: produceChunkAsyncWithFailures , doesnt allow passing in serializers , it expects a `ByteRecord` aka `ProducerRecord[Array[Byte],Array[byte]]` .
I was trying to converge on one method as the core method and everything else is the equivalent of a better interface for a particular task , as this is the way it is currently implemented.
So the serialization needs to be handled in this core method.
(the underlying zio library actually does the same thing, everything is reduced down to produceChunkAsyncWithFailures and serialization is called beforehand

**Comment by yisraelU** [comment]
> you'd think that such an API change would get a minor version update...

absolutely !!

**Comment by soujiro32167** [comment]
Given there are no better ways to compose producers, I don't see a way to improve here...
@calvinlfer @anakos what do you think?
I'm going to open a ticket on zio-kafka to create composition functions, so this kind of middleware is easier

**Comment by soujiro32167** [comment]
https://github.com/zio/zio-kafka/issues/1039

You have my thanks and 👍

**Comment by yisraelU** [comment]
> Given there are no better ways to compose producers, I don't see a way to improve here...
> @calvinlfer @anakos what do you think?
> I'm going to open a ticket on zio-kafka to create composition functions, so this kind of middleware is easier

A side note , I'm not sure if the whole nested effect thing is a good interface for a public api for modeling 2 layers of async

**Comment by calvinlfer** [comment]
Superseded by https://github.com/kaizen-solutions/trace4cats-zio-extras/pull/32

**Pull request closed without merging**